instance/api/instance.py

```diff
@@ -70,6 +70,14 @@ def get_view_name(self):
         Get the verbose name for each view
         """
         suffix = self.suffix
-        if self.action == 'retrieve':
+        # DRF will only set the `.action` attribute on a view in the context of a request.
+        # In recent versions of DRF, the `get_breadcrumbs` function instantiates a view
+        # and calls `get_view_name` on it outside of the context of a request,
+        # cf. https://github.com/tomchristie/django-rest-framework/pull/3273.
+        # In that case, `.action` is not set on the view, so referencing it directly
+        # (via `self.action`) inside this method will fail with an `AttributeError`.
+        # We guard against that here by using `getattr` to retrieve the value
+        # of the `action` field.
+        if getattr(self, 'action', None) == 'retrieve':
             suffix = "Details"
         return "Instance {}".format(suffix)
```

Comment:
@antoviaque We change the view name to get a better/more specific heading displayed in the browsable API. When overriding `get_view_name`, we get "Instance List" on `/api/v1/instance/`, and "Instance Details" on `/api/v1/instance/{pk}/`:

* ![instance-list-custom-view-name](https://cloud.githubusercontent.com/assets/961441/20178327/e6fa4bca-a750-11e6-9171-34e01fbb9b9e.png)

* ![instance-details-custom-view-name](https://cloud.githubusercontent.com/assets/961441/20178331/eb48b48c-a750-11e6-9637-3979afa05688.png)

When *not* overriding `get_view_name`, we get "Instance List" on `api/v1/instance/` (same as before), but on `/api/v1/instance/{pk}/` we get "Instance Instance":

* ![instance-list-default-view-name](https://cloud.githubusercontent.com/assets/961441/20178402/427d1252-a751-11e6-9111-933c5f531af4.png)

* ![instance-details-default-view-name](https://cloud.githubusercontent.com/assets/961441/20178403/47e8aef4-a751-11e6-9bc2-7abf4606ac0a.png)

I'm not that familiar with DRT, so I'm not sure if there is a broader fix/refactoring that we could apply here (if there is, it would probably also take care of the breadcrumbs issue that I laid out in the PR description).